### ISSUE TITLE
'singular matrix' error for perfect correlations

### ISSUE THREAD

**Issue by eirol** [tadamcz/copula-wrapper#1]
#### Description

When there is perfect correlation between two variables (in the simplest case I've tried, at least), JointDistribution fails with a "singular matrix" error.

#### To reproduce

```python
from copula_wrapper import JointDistribution
from scipy import stats

marginals = {
 'a': stats.uniform(0.75, 3),
 'b': stats.uniform(0.75, 3),
}

pairwise_rank_corr = {
  ('a', 'b'): 1,
}

joint = JointDistribution(marginals, pairwise_rank_corr, rank_corr_method = "spearman")
```

#### Stack trace
```
Traceback (most recent call last):
  File "test.py", line 13, in <module>
    joint = JointDistribution(marginals, pairwise_rank_corr, rank_corr_method = "spearman")
  File "joint_distribution.py", line 60, in __init__
    copula_instance = GaussianCopula(corr=pearsons_rho)
  File "/usr/lib/python3.10/site-packages/statsmodels/distributions/copula/elliptical.py", line 153, in __init__
    self.distr_mv = stats.multivariate_normal(cov=corr)
  File "/usr/lib/python3.10/site-packages/scipy/stats/_multivariate.py", line 360, in __call__
    return multivariate_normal_frozen(mean, cov,
  File "/usr/lib/python3.10/site-packages/scipy/stats/_multivariate.py", line 730, in __init__
    self.cov_info = _PSD(self.cov, allow_singular=allow_singular)
  File "/usr/lib/python3.10/site-packages/scipy/stats/_multivariate.py", line 165, in __init__
    raise np.linalg.LinAlgError('singular matrix')
numpy.linalg.LinAlgError: singular matrix
```

**Comment by tadamcz** [comment]
This was a limitation of statsmodels. But parameter allow_singular been added in statsmodels 0.14: https://github.com/statsmodels/statsmodels/issues/8454

**Comment by tadamcz** [comment]
I've added this param to the wrapper as well in https://github.com/tadamcz/copula-wrapper/commit/92ee9ae65eefa06ef0cd1614c8b87f90c15d81a3 🚀 

Published to pypi as 0.1.2.

Please note the warning from statsmodels:

```
The behavior when the correlation 
matrix is singular is determined by  ``scipy.stats.multivariate_normal`` and might not
be appropriate for all methods. Behavior might change in future versions. 
```

More detail at this PR: https://github.com/statsmodels/statsmodels/pull/8504
